### ISSUE TITLE
Pass in array of data with labels seperately + a few questions

### ISSUE THREAD

**Issue by inspire22** [ankane/eps#28]
Thanks for this project! I'm enough of a fan I look at your repository list to see what else I should be playing with lol.

1) In the readme it says "Data can be an array of arrays" - but then the "y" passed in is also numerical - should this be a list of feature names, or how could I go about populating the model with a list of values (from rails 'pluck') and a list of feature names in order.

2) Should I consider converting my ruby data to a data frame, or will EPS handle converting it into the most efficient format to pass to lightBGM?

3) When deploying a model to production, do I need to run build with Eps::Model.new(data, split: false), or will both the test & train data be included in the final model automatically?

4)  Bayes classification seems pretty slow.  If you wanted, I've had a good experience with this gem for very fast bayesian classification in ruby https://github.com/Coolnesss/fast-bayes (I could be wrong it being that slow, I'd been using a slow TextStat.dale_chall_readability_score which I removed at the same time)

5) A potential future metric - My favorite method of text classification these days is generating LLM embeddings then using KNN averaging to choose a score/class.  Sounds hard to include in a project like this since you need a LLM and vector database but thought I'd mention it.

6) The link in the Readme for "[mitigate bias](http://aif360.mybluemix.net/) in training data" is broken, was wondering if this would help mitigate the fact I have more 'low' and 'high' labeled data than 'middle'.

**Comment by ankane** [comment]
Hi @inspire22,

1. Feature names are auto-generated if you pass an array of arrays (you'll need to use a different format to pass your own feature names).
2. I haven't benchmarked the different input formats, so not sure which is most efficient.
3. You'll need to retrain the model with all of the data (it doesn't happen automatically).
4. Might be worth trying with YJIT to see if it helps.
5. Thanks for sharing, but think it's outside the scope of the project.
6. Thanks, will fix the link. It might help if the dataset is very imbalanced, but in general, I'd follow [this advice](https://developers.google.com/machine-learning/crash-course/overfitting/imbalanced-datasets).
